src/app.rs

```diff
@@ -52,6 +56,8 @@ pub struct AppContext {
     pub storage: Arc<Storage>,
     // Cache instance for the application
     pub cache: Arc<cache::Cache>,
+    /// Storage for any helpful app-specific objects
+    pub extensions: Arc<RwLock<Extensions>>,
```

Comment:
Is it possible to access Extensions directly from Axum instead of from the HTTP request?

Reply:
It must implement Clone there is no other way but most of the time would actually have something like `Arc<InnerService>` so the overhead would be cloning an `Arc` which isn't that big.

Reply:
On the first, I'm not sure what you mean. The Extensions struct comes from the HTTP crate, since it does most of what this needs and is defined there. Nothing from this comes from any particular HTTP request though.

On the second, yes that's true, and I used an `Arc` for it, as @DenuxPlays said, which has cheap clones. I suppose it's on the app author to not put anything in there that is expensive to clone.

Reply:
Thats Fine for me.

For the http issue we could reuse the DashMap Part but instead of (TypeId, String) just use TypeId for the key.

In this way we would be Independent from the http crate.

Another Option would be to just copy the Extensions Code from the http crate if the license allows it.

Reply:
@DenuxPlays Sounds good! I'm working on it right now and will open a PR soon. It would be great to get your code review!

Reply:
Oh I updated my pr to reflect this.
Also simplified it a lot as the whole service can be moved in another crate.

You can take a look and maybe use it as a base or something

Reply:
Finished my pr by adding docs.

However there is another thing I think we forgot.
I do like the change that `init_logger` now has access to `AppContext` but I think we should do that in a separate pr for cleaner Changelogs and git history.